SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialShadowActor.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+#include "SpatialConstants.h"
+#include "SpatialShadowActor.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialShadowActor, Log, All);
+
+UCLASS(Transient)
+class SPATIALGDK_API USpatialShadowActor : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	USpatialShadowActor(const FObjectInitializer& ObjectInitializer = FObjectInitializer::Get());
+
+	void Init(const Worker_EntityId InEntityId, AActor* InActor);
+
+	void CheckUnauthorisedDataChanges(const Worker_EntityId InEntityId, AActor* InActor);
+
+	void Update(const Worker_EntityId InEntityId, AActor* InActor);
+
+private:
+	Worker_EntityId EntityId;
+	AActor* Actor;
```

Comment:
Not a UPROPERTY or weak pointer?

Reply:
If true, is there any reason why this class needs to be an UObject?

Reply:
+1

Reply:
Already using the ActorSystem to add, update and remove spatial shadow actors.
However, Nic made a good point that we can decouple from the replication.
I have refactored to no longer override the BuildConsiderList and check for changes in the SpatialNetDriver::Tick() instead.
Reverted all Engine changes.